pkgs/development/python-modules/pysdl3/default.nix

```diff
@@ -0,0 +1,102 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  fetchFromGitHub,
+  python,
+  buildPythonPackage,
+  setuptools-scm,
+  packaging,
+  aiohttp,
+  requests,
+
+  # native dependencies
+  sdl3,
+  sdl3-ttf,
+  sdl3-image,
+}:
+
+let
+  dochash =
+    if stdenv.hostPlatform.isLinux then
+      "sha256-+1zLd308zL+m68kLMeOWWxT0wYDgCd6g9cc2hEtaeUs="
+    else if stdenv.hostPlatform.isDarwin then
+      "sha256-2uB9+ABgv5O376LyHb0ShGjM4LHYzMRMxk/k+1LBmv0="
+    else if stdenv.hostPlatform.isWindows then
+      "sha256-46bQSPYctycizf2GXichd5V74LjxwIAPhBmklXAJ/Jg="
+    else
+      throw "PySDL3 does not support ${stdenv.hostPlatform.uname.system}";
+  lib_ext = stdenv.hostPlatform.extensions.sharedLibrary;
+in
+buildPythonPackage rec {
+  pname = "pysdl3";
+  version = "0.9.8b1";
+  pyproject = true;
+
+  pythonImportsCheck = [ "sdl3" ];
+
+  src = fetchFromGitHub {
+    owner = "Aermoss";
+    repo = "PySDL3";
+    tag = "v${version}";
+    hash = "sha256-FVUCcqKTq6qdNkYHTYFiUxt2HIaNC5LK0BEUfz8Mue8=";
+  };
+
+  docfile = fetchurl {
+    url = "https://github.com/Aermoss/PySDL3/releases/download/v${version}/${stdenv.hostPlatform.uname.system}-Docs.py";
+    hash = "${dochash}";
+  };
```

Comment:
rather than fetching the generated assets, please generate the assets in the derivation if possible. The procedure seems to be in https://github.com/Aermoss/PySDL3/blob/main/.github/workflows/generate_docs.yml.

If we do have to fetch it, then we should set `meta.sourceProvenance` accordingly.

Reply:
The generator attempts to first download the file from the github release so that would not really change anything. If downloading it from the github release fails, it starts scraping docs from libsdl.org which I'm also not a huge fan of (it mostly seems like a great way to get our CI machines IP blocked these days).

Ideally we'd generate the docs from our own SDL3 build. I was going to propose supporting that case to upstream but didn't manage to get nixpkgs SDL3 to generate its documentation...

Reply:
What shoud `meta.sourceProvenance` be set to in this case? `binaryBytecode`?